### PR TITLE
Split API changes into individual pages

### PR DESCRIPTION
**Pull request by dstansby**
This gets rid of the huge API changes page, and splits it into individual files for each release.

- https://904-74355158-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/api/api_changes.html is the new main page, which contains the most recent API changes
- This links to https://904-74355158-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/api/api_changes_old.html , which contains a list of each individual API change page.

### CODE REVIEW

File: doc/api/api_changes.rst [hunk 1]
```diff
@@ -19,8 +26,8 @@ For new features that were added to Matplotlib, please see
 
    next_api_changes/*
```
[the remark this answers is not in the document]
Comment (timhoffm) [reply doc/api/api_changes.rst]: Couldn't we just write all the API changes in a single file? I think this was proposed in another thread. Might be a problem though with getting many merge conflicts.

Alternatively, collect the unmerged stuff on a separate page. Or at least Give the list of unmerged stuff a section title and comment or a `.. note::` block. The current state is a bit confusing for other developers not familiar with the process (e.g. me) and for people accidentially ending in devdocs.

[the remark this answers is not in the document]
Comment (jklymak) [reply doc/api/api_changes.rst]: Yes, the problem is too many conflicts if there is one file.  @tacaswell has a script that merges them all up....

[the remark this answers is not in the document]
Comment (dstansby) [reply doc/api/api_changes.rst]: I've added a note to hopefully explain things a bit better.

**Comment by dstansby** [comment]
👍 with two approvals I'm going to merge

**Pull request merged**